exonum/src/blockchain/service.rs

```diff
@@ -400,19 +402,39 @@ impl SharedNodeState {
 
         lock.peers_info.clear();
         lock.majority_count = state.majority_count();
+        lock.is_validator = state.is_validator();
+        lock.validators.clear();
 
-        for (p, c) in state.peers().iter() {
+        for (p, c) in state.peers() {
             lock.peers_info.insert(c.addr(), *p);
         }
+        for v in state.validators() {
```

Comment:
Maybe rewrite it to something like this? (then we won't have to clean validators list either)
```
lock.validators = state.validators().to_vec();
```